platforms/chibios/drivers/spi_master.c

```diff
@@ -267,17 +318,30 @@ spi_status_t spi_write(uint8_t data) {
 
 spi_status_t spi_read(void) {
     uint8_t data = 0;
+
+#if defined(QMK_MCU_SERIES_STM32H7XX)
+    SCB_InvalidateDCache_by_Addr((uint32_t *)(((uint32_t)&data) & ~(uint32_t)0x1F), 1 + 32);
+#endif
+
     spiReceive(&SPI_DRIVER, 1, &data);
 
     return data;
 }
 
 spi_status_t spi_transmit(const uint8_t *data, uint16_t length) {
+#if defined(QMK_MCU_SERIES_STM32H7XX)
+    SCB_CleanDCache_by_Addr((uint32_t *)(((uint32_t)data) & ~(uint32_t)0x1F), length + 32);
+#endif
+
     spiSend(&SPI_DRIVER, length, data);
     return SPI_STATUS_SUCCESS;
 }
 
 spi_status_t spi_receive(uint8_t *data, uint16_t length) {
+#if defined(QMK_MCU_SERIES_STM32H7XX)
+    SCB_InvalidateDCache_by_Addr((uint32_t *)(((uint32_t)data) & ~(uint32_t)0x1F), length + 32);
+#endif
```

Comment:
Again, because the buffer might not be aligned to a cache line, and might share some cache lines with unrelated data, this should probably be `CleanInvalidate` instead of just `Invalidate`.